src/AddIns/Uno.UI.Lottie/LottieVisualSource.cs

```diff
@@ -133,22 +146,72 @@ Size IAnimatedVisualSource.Measure(Size availableSize)
 			var availableWidth = availableSize.Width;
 			var availableHeight = availableSize.Height;
 
+			var resultSize = availableSize;
+
 			if (double.IsInfinity(availableWidth))
 			{
 				if (double.IsInfinity(availableHeight))
 				{
 					return compositionSize;
 				}
 
-				return new Size(availableHeight * compositionSize.Width / compositionSize.Height, availableHeight);
+				resultSize = new Size(availableHeight * compositionSize.Width / compositionSize.Height, availableHeight);
 			}
 
 			if (double.IsInfinity(availableHeight))
 			{
-				return new Size(availableWidth, availableWidth * compositionSize.Height / compositionSize.Width);
+				resultSize = new Size(availableWidth, availableWidth * compositionSize.Height / compositionSize.Width);
+			}
+
+			InnerMeasure(resultSize);
+
+			return resultSize;
+		}
+
+		partial void InnerMeasure(Size size);
+
+		private bool TryLoadEmbeddedJson(Uri uri, out string json)
+		{
+			if (uri.Scheme != "embedded")
+			{
+				json = null;
+				return false;
+			}
+
+			var assemblyName = uri.Host;
+
+			Assembly assembly;
+			if (assemblyName == ".")
+			{
+				assembly = Application.Current.GetType().Assembly;
+			}
+			else
+			{
+				assembly = Assembly.Load(assemblyName);
+			}
+
+			if (assembly == null)
+			{
+				json = null;
+				return false;
+			}
+
+			var resourceName = uri.AbsolutePath.Substring(1).Replace("(assembly)", assembly.GetName().Name);
+			using var stream = assembly.GetManifestResourceStream(resourceName);
+			if (stream == null)
+			{
+				if (this.Log().IsEnabled(LogLevel.Warning))
+				{
+					this.Log().Warn($"Unable to find embedded resource named '{resourceName}' to load.");
+				}
+				json = null;
+				return false;
 			}
 
-			return availableSize;
+			var bytes = new byte[(int)stream.Length];
+			stream.Read(bytes, 0, bytes.Length);
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Check the return value of the 'Read' call to see how many bytes were read.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=5477430)